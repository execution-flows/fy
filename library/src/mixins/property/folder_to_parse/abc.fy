from pathlib import Path


property folder_to_parse: Path
