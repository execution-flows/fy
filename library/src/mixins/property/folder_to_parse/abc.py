import abc

from pathlib import Path


class With_FolderToParse_PropertyMixin_ABC(abc.ABC):
    @property
    @abc.abstractmethod
    def _folder_to_parse(self) -> Path:
        raise NotImplementedError()
