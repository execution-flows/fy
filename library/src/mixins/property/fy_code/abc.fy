property fy_code: str
