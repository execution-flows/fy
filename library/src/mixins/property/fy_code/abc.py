import abc


class With_FyCode_PropertyMixin_ABC(abc.ABC):
    @property
    @abc.abstractmethod
    def _fy_code(self) -> str:
        raise NotImplementedError()
