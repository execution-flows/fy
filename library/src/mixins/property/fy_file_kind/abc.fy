from domain.parsed_fy_py_file import ParsedFyPyFileKind


property fy_file_kind: ParsedFyPyFileKind
