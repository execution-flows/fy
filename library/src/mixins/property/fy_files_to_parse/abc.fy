from pathlib import Path
from typing import List


property fy_files_to_parse: List[Path]
