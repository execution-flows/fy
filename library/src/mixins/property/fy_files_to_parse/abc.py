from pathlib import Path
from typing import List

import abc


class With_FyFilesToParse_PropertyMixin_ABC(abc.ABC):
    @property
    @abc.abstractmethod
    def _fy_files_to_parse(self) -> List[Path]:
        raise NotImplementedError()
