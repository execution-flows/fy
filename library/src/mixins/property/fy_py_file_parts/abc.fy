from domain.parsed_fy_py_file import FyPyFileParts


property fy_py_file_parts: FyPyFileParts
