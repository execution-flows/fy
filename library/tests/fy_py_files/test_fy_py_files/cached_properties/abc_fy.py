"""fy
property greeting: str
"""

import abc


# fy:start ===>>>
class Greeting_PropertyMixin_ABC(abc.ABC):
    @property
    @abc.abstractmethod
    def _greeting(self) -> str:
        raise NotImplementedError()
        # fy:end <<<===
