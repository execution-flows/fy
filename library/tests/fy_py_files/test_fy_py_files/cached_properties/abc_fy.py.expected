"""fy
property greeting: str
"""
import abc


# fy:start <<<===
class With_Greeting_PropertyMixin_ABC(abc.ABC):
    @property
    @abc.abstractmethod
    def _greeting(self) -> str:
        raise NotImplementedError()
        # fy:end <<<===
